File: LC_Gen_Codes/2644_setting.py
class Solution:
     def maxDivScore(self, nums: List[int], divisors: List[int]) -> int:
         score = [0] * len(divisors)  # initialize score array with zeros
         for num in nums:
             for i, divisor in enumerate(divisors):
                 if num % divisor == 0:
                     score[i] += 1  # increment the score if num is divisible by divisor
         max_score = max(score)  # find the maximum score
         max_score_divisors = [divisors[i] for i, s in enumerate(score) if s == max_score]
         return min(max_score_divisors)  # return the minimum of the divisors with maximum score

File: LC_Gen_Codes_Java/2644_setting.java
class Solution {
    public int maxDivScore(int[] nums, int[] divisors) {
        int[] score = new int[divisors.length]; // initialize score array with zeros
        for (int num : nums) {
            for (int i = 0; i < divisors.length; i++) {
                if (num % divisors[i] == 0) {
                    score[i] += 1; // increment the score if num is divisible by divisor
                }
            }
        }
        int max_score = Integer.MIN_VALUE;
        for (int s : score) {
            max_score = Math.max(max_score, s); // find the maximum score
        }
        List<Integer> max_score_divisors = new ArrayList<>();
        for (int i = 0; i < score.length; i++) {
            if (score[i] == max_score) {
                max_score_divisors.add(divisors[i]);
            }
        }
        int min_divisor = Integer.MAX_VALUE;
        for (int divisor : max_score_divisors) {
            min_divisor = Math.min(min_divisor, divisor); // find the minimum of the divisors with maximum score
        }
        return min_divisor;
    }
}